descwl_shear_sims/trivial_sim.py

```diff
@@ -250,3 +318,1004 @@ def make_wcs(*, scale, image_origin, world_origin, theta=None):
+        self._wcs = wcs
+
+    def __call__(self, *, x, y, center_psf, get_offset=False):
+        """
+        center_psf is ignored
+        """
+        image_pos = galsim.PositionD(x=x, y=y)
+
+        offset = copy.deepcopy(self._offset)
+
+        if center_psf:
+            print("ignoring request to center psf, using internal offset")
+
+        gsimage = self._psf.drawImage(
+            nx=self._psf_dim,
+            ny=self._psf_dim,
+            offset=offset,
+            wcs=self._wcs.local(image_pos=image_pos),
+        )
+        if get_offset:
+            if offset is None:
+                offset = galsim.PositionD(x=0.0, y=0.0)
+            return gsimage, offset
+        else:
+            return gsimage
+
+
+def make_seobs(
+    *,
+    rng,
+    band,
+    noise,
+    objlist,
+    shifts,
+    dim,
+    psf,
+    psf_dim,
+    dither=False,
+    rotate=False,
+    bright_objlist=None,
+    bright_shifts=None,
+    bright_mags=None,
+    mask_threshold=None,
+    cosmic_rays=False,
+    bad_columns=False,
+    star_bleeds=False,
+):
+    """
+    make a grid sim with trivial pixel scale, fixed sized
+    exponentials and gaussian psf
+
+    Parameters
+    ----------
+    rng: numpy.random.RandomState
+        The random number generator
+    noise: float
+        Gaussian noise level
+    objlist: list
+        List of GSObj
+    shifts: array
+        Array with fields dx and dy, which are du, dv offsets
+        in sky coords.
+    dim: int
+        Dimension of image
+    psf: GSObject or PowerSpectrumPSF
+        the psf
+    psf_dim: int
+        Dimensions of psf image that will be drawn when psf func is called
+    dither: bool
+        If set to True, dither randomly by a pixel width
+    rotate: bool
+        If set to True, rotate the image randomly
+    cosmic_rays: bool
+        If True, put in cosmic rays
+    bad_columns: bool
+        If True, put in bad columns
+    """
+
+    dims = [dim]*2
+    cen = (np.array(dims)-1)/2
+
+    se_origin = galsim.PositionD(x=cen[1], y=cen[0])
+    if dither:
+        dither_range = 0.5
+        off = rng.uniform(low=-dither_range, high=dither_range, size=2)
+        offset = galsim.PositionD(x=off[0], y=off[1])
+        se_origin = se_origin + offset
+    else:
+        offset = None
+
+    if rotate:
+        theta = rng.uniform(low=0, high=2*np.pi)
+    else:
+        theta = None
+
+    se_wcs = make_wcs(
+        scale=SCALE,
+        theta=theta,
+        image_origin=se_origin,
+        world_origin=WORLD_ORIGIN,
+    )
+
+    convolved_objects, psf_gsobj = get_convolved_objects(
+        objlist=objlist,
+        psf=psf,
+        shifts=shifts,
+        offset=offset,
+        se_wcs=se_wcs,
+        se_origin=se_origin,
+    )
+
+    objects = galsim.Add(convolved_objects)
+
+    # everything gets shifted by the dither offset
+    image = objects.drawImage(
+        nx=dim,
+        ny=dim,
+        wcs=se_wcs,
+        offset=offset,
+    )
```

Comment:
When you do this, galsim always calls `.local` on the input WCS and then draws with that jacobian.

Reply:
wtf?  what's the point of having complicated wcs support if it is not used?